x-pack/plugins/asset_manager/server/lib/validators/validation_error.ts

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+interface ErrorOptions {
+  statusCode?: number;
+}
+
+export class AssetsValidationError extends Error {
+  public statusCode: number;
+
+  constructor(message: string, { statusCode = 500 }: ErrorOptions = {}) {
```

Comment:
(http semantics intensifies) should this default to 400 ?

Reply:
lol, I was thinking about this like a ResponseError but specifically as a ValidationError, I think you're right.